src/renderer/src/routes/modals/PersistContact/index.tsx

```diff
@@ -0,0 +1,194 @@
+    if (addressToEdit) {
+      addressToEdit.address = address.address
+      addressToEdit.blockchain = address.blockchain
+    } else {
+      setAddresses(prev => [...prev, address])
+    }
+  }
+
+  const handleDeleteAddress = (addressToDelete: TContactAddress) => {
+    if (!contact) return
+    const filterAddress = addresses.filter(address => address !== addressToDelete)
+
+    setAddresses(filterAddress)
+  }
+
+  const openAddAddressModal = (selectedAddress?: TContactAddress) => {
+    if (!contactName) {
+      form.setError('name', { message: t('invalidName') })
+      return
+    }
+
+    modalNavigate('addAddressStep1', {
+      state: { contactName: contactName, address: selectedAddress || null, handleAddAddress: handleAddAddress },
+    })
+  }
+
+  const handleDeleteContact = (contact: IContactState) => {
+    dispatch(contactReducerActions.deleteContact(contact.id))
+    modalNavigate(-1)
+  }
+
+  const handleSubmit: SubmitHandler<TFormData> = async data => {
+    if (!data.name.length) {
+      form.setError('name', { message: t('invalidName') })
+      return
+    }
+
+    if (!addresses.length) {
+      setShowAddressError(true)
+      return
+    }
+
+    try {
+      if (contact) {
+        dispatch(contactReducerActions.saveContact({ name: contactName, addresses: addresses, id: contact.id }))
+      } else {
+        const newContact = new Contact({ name: contactName, addresses: addresses })
+        dispatch(contactReducerActions.saveContact(newContact))
+      }
+
+      modalNavigate(-1)
+    } catch (e) {
+      console.error(e)
+    }
+  }
+
+  return (
+    <ModalLayout
+      heading={contact ? t('editContact') : t('addContact')}
+      headingIcon={<TbPlus />}
+      headingIconFilled={false}
+    >
+      <form onSubmit={form.handleSubmit(handleSubmit)} className="flex flex-col justify-between h-full">
+        <div>
+          <div className="mb-5">
+            <div className="text-gray-100 font-bold font-bold pb-2">{t('name')}</div>
+            <Input
+              placeholder={t('enterAName')}
+              {...form.register('name')}
+              errorMessage={form.formState.errors.name?.message}
+            />
+          </div>
+
+          <div className="mb-5">
+            <div className="text-gray-100 font-bold pb-2">{t('addresses')}</div>
+            {addresses &&
+              addresses.map((address, index) => (
+                <div key={index} className="flex items-center h-12 rounded bg-asphalt w-full pr-3 mb-2 justify-between">
+                  <Button
+                    clickableProps={{ className: 'border-none' }}
+                    onClick={() => openAddAddressModal(address)}
+                    label={address.address}
+                    leftIcon={<BlockchainIcon blockchain={address.blockchain} type="white" />}
+                    flat
+                    variant="outlined"
+                    className="truncate"
+                    type="button"
+                  />
```

Comment:
It does not need to be a button. I think its not possible to edit an added address